cmd/agent/common/autodiscovery.go

```diff
@@ -103,6 +113,20 @@ func setupAutoDiscovery(confSearchPaths []string, metaScheduler *scheduler.MetaS
 			listeners = append(listeners, config.Listeners{Name: name})
 		}
 
+		// The "docker" listener was replaced with the "container" one that
+		// supports Docker, but also other runtimes. We need this conversion to
+		// avoid breaking configs that included "docker".
+		dockerIdx := -1
```

Comment:
Docker could be present multiple times, I think that something like:
```
for idx := range listeners {
  if listeners[idx].Name == "docker" {
    listeners[idx].Name = "container"
  }
}
```

Would be easier and work

Reply:
Changed :+1: